### PR TITLE
Added missing Google Hardware Attestation Root certificate

### PR DESCRIPTION
**Pull request by dipskp**
The code base had one Google root certificate, which has been used to sign the root  certificate in a real attestation certificate chain from some compliant devices. Some devices will fail this check because there are two certs which are used to sign the root cert.
The missing one has been added

More details : https://developer.android.com/training/articles/security-key-attestation

### CODE REVIEW

**Comment by eranmes** [comment]
Thanks for taking the time to put together a pull request adding the second Google root.

Could you please update this PR to _only_ include the second root, without any formatting changes or code re-arrangement?

**Comment by dipskp** [comment]
> Thanks for taking the time to put together a pull request adding the second Google root.
> 
> Could you please update this PR to _only_ include the second root, without any formatting changes or code re-arrangement?

Just to clarify, I had formatted code earlier but in next commit, I have added the second root in Constants and its check in KeyAttestationExample under verifyCertificateChain(). Should I keep the check?

**Comment by dipskp** [comment]
Hi @eranmes, thanks for the feedback, I have updated the PR, can you please review the changes?

https://github.com/google/android-key-attestation/pull/7/commits/3dbafbc937a91b52c2b72b7db5e98bfcb6763854

**Comment by dipskp** [comment]
Hi @eranmes, because of your busy schedule you might have missed my request. Can you please take a look and respond.
Awaiting your action.

Thanks

File: server/src/main/java/com/android/example/KeyAttestationExample.java [hunk 1]
```diff
@@ -249,13 +250,28 @@ private static void verifyCertificateChain(X509Certificate[] certs)
...
-              + " software, not in secure hardware. This means that, although the attestation"
-              + " contents are probably valid and correct, there is no proof that they are in fact"
-              + " correct. If you're using a production-level system, you should now treat the"
-              + " properties of this attestation certificate as advisory only, and you shouldn't"
-              + " rely on this attestation certificate to provide security guarantees.");
+      X509Certificate secureRoot =
+          (X509Certificate)
+              CertificateFactory.getInstance("X.509")
+                  .generateCertificate(
+                      new ByteArrayInputStream(GOOGLE_ROOT_CERTIFICATE.getBytes(UTF_8)));
```
Comment (veehaitch) [inline server/src/main/java/com/android/example/KeyAttestationExample.java:257]: That's supposed to reference the newly added CA, isn't it?

```suggestion
                      new ByteArrayInputStream(GOOGLE_ROOT_CERTIFICATE_2.getBytes(UTF_8)));
```

Would be great to also have this PR include a test attestation certificate chain which roots in `GOOGLE_ROOT_CERTIFICATE_2`.

**Review by veehaitch — CHANGES_REQUESTED** [review]

**Comment by eranmes** [comment]
Apologies for the delay in reviewing, below are the suggestions needed to get this pull request merged:
* Please only have one commit in the CL introducing the new root (so we don't have commits that change formatting and then change it back; it muddles the code history and 'blame' layer).
* There's code duplication for checking the first and second roots. We can just extract both public keys and compare the input to them both, rather than duplicating lines 239-251 again.
* As @veehaitch suggested, please include a test that contains an attestation record signed by the root you're adding.

File: server/src/main/java/com/android/example/KeyAttestationExample.java (hunk 1 above)
Comment (dipskp) [reply server/src/main/java/com/android/example/KeyAttestationExample.java:257]: @veehaitch yes GOOGLE_ROOT_CERTIFICATE_2 is second one . I will also add the test cert chain for the 2nd certificate.

**Comment by eranmes** [comment]
I'm closing this PR because Shawn has landed a change that obsoletes this: Shawn's change is comparing the root's public key, rather than the root certificate itself.

**Pull request closed without merging**